### PR TITLE
feat(Set): Add `Set.Finite.powerset`

### PR DESCRIPTION
**Pull request by SnO2WMaN**
If set `s` is finite, then powerset of `s` (`𝒫 s`) is finite.

Zulip: https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Set.20is.20finite.2C.20then.20its.20powerset.20is.20finite/near/447338619

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Finite.lean [hunk 1]
```diff
@@ -1221,6 +1221,9 @@ theorem Finite.induction_to_univ [Finite α] {C : Set α → Prop} (S0 : Set α)
     (H0 : C S0) (H1 : ∀ S ≠ univ, C S → ∃ a ∉ S, C (insert a S)) : C univ :=
   finite_univ.induction_to S0 (subset_univ S0) H0 (by simpa [ssubset_univ_iff])
 
+protected theorem Finite.powerset {s : Set α} (h : s.Finite) : (𝒫 s).Finite :=
+  Set.Finite.finite_subsets h
```
Comment (fpvandoorn) [inline Mathlib/Data/Set/Finite.lean:1225]:
```suggestion
  h.finite_subsets
```

**Review by fpvandoorn — COMMENTED** [review]
Please move this directly below `Finite.finite_subsets`

**Comment by fpvandoorn** [comment]
Ah, I didn't know you hadn't contributed to Mathlib when I made the request. We have a bit of an unusual contribution workflow, where contributors make PRs from branches of the main repository (not their own fork). Only then our CI (continuous integration) will work correctly.

I gave you write permission to branches. Could you please re-open this PR from a branch of the main repository

More information can be found here: https://leanprover-community.github.io/contribute/index.html

**Comment by SnO2WMaN** [comment]
@fpvandoorn I reopened with #14175 . Sorry.

**Comment by fpvandoorn** [comment]
Thanks!

**Pull request closed without merging**